pkg/controller/schedulingmanager/controller.go

```diff
@@ -113,31 +113,33 @@ func (c *SchedulerController) Run(stopChan <-chan struct{}) {
 func (c *SchedulerController) reconcile(qualifiedName util.QualifiedName) util.ReconciliationStatus {
 	key := qualifiedName.String()
 
-	glog.Infof("Running reconcile FederatedTypeConfig for %q", key)
+	glog.V(3).Infof("Running reconcile FederatedTypeConfig for %q", key)
 
 	schedulingType := schedulingtypes.GetSchedulingType(qualifiedName.Name)
 	if schedulingType == nil {
 		// No scheduler supported for this resource
 		return util.StatusAllOK
 	}
-	if c.runningPlugins.Has(qualifiedName.Name) {
-		// Scheduler and plugin are already running
-		return util.StatusAllOK
-	}
 
 	cachedObj, exist, err := c.store.GetByKey(key)
 	if err != nil {
 		runtime.HandleError(fmt.Errorf("Failed to query FederatedTypeConfig store for %q: %v", key, err))
 		return util.StatusError
 	}
+
 	if !exist {
+		c.stopScheduler(schedulingType.Kind, qualifiedName)
 		return util.StatusAllOK
 	}
+
 	typeConfig := cachedObj.(*corev1a1.FederatedTypeConfig)
+	if typeConfig.Spec.PropagationEnabled == false || typeConfig.DeletionTimestamp != nil {
+		c.stopScheduler(schedulingType.Kind, qualifiedName)
+		return util.StatusAllOK
+	}
 
-	deleted := typeConfig.DeletionTimestamp != nil
-	if deleted {
-		// Do not support deletion yet
+	if c.runningPlugins.Has(qualifiedName.Name) {
```

Comment:
(No action required) I'm not sure why management of running plugins needs to be performed outside of the scheduling preference controller.   Or that the scheduling preference controller needs to be started separately from the plugin.  I think this suggests the possibility of simplification in future iterations of this area of functionality.

Reply:
Agree. This is also what I'm thinking when change the stop management code to the related module.